src/util/system.cpp

```diff
@@ -506,14 +506,46 @@ std::string ArgsManager::GetHelpMessage() const
     return usage;
 }
 
+std::string ArgsManager::GetSpecificHelpMessage()
+{
+    const std::string helpParams[] = {"-?", "-h", "-help", "-help-debug"};
+    std::string helpName = "";
+    for (long unsigned int i = 0; i < sizeof(helpParams); ++i) {
+        if (this->IsArgSet(helpParams[i])) {
+            helpName = this->GetArg(helpParams[i], "");
+            break;
+        }
+    }
+    // Return unspecific help if no arg was given
+    if (helpName == "")
+        return this->GetHelpMessage();
+
+    std::pair<std::string, ArgsManager::Arg> command;
+    bool cmdExists = false;
+    LOCK(cs_args);
+    for (const auto& arg_map : m_available_args) {
+        for (auto const& arg : arg_map.second) {
+            if (arg.first == "-" + helpName) {
+                command = arg;
+                cmdExists = true;
+                break;
+            }
+        }
+    }
+    // Return the full help if the command is not existent
```

Comment:
It should be an error IMO if the command is not existent. The "in case of doubt dump the manual in the users face" behavior frequently frustrates me in other software :laughing: